clang/lib/3C/CastPlacement.cpp

```diff
@@ -57,9 +57,19 @@ bool CastPlacementVisitor::VisitCallExpr(CallExpr *CE) {
       Expr *ArgExpr = A;
       if (FD && PIdx < FD->getNumParams()) {
         const int TyVarIdx = FV->getExternalParam(PIdx)->getGenericIndex();
+        // Check if local type vars are available
         if (TypeVars.find(TyVarIdx) != TypeVars.end() &&
-            TypeVars[TyVarIdx] != nullptr)
-          TypeVar = TypeVars[TyVarIdx];
+            TypeVars[TyVarIdx].first != nullptr) {
```

Comment:
If the `TypeVars` value pair is made a struct, then maybe some of this logic could be moved into a method of the struct.